### PR TITLE
Propogate error when docsbuild fails, add docs build to CI.

### PR DESCRIPTION
**Pull request by alliepiper**
Also fixes CI so that we don't run jobs when only `docs/` is modified.

### CODE REVIEW

**Review by miscco — APPROVED** [review]

**Review by miscco — APPROVED** [review]

**Comment by alliepiper** [comment]
Docs build is working: 

- Job: https://github.com/NVIDIA/cccl/actions/runs/9487515807/job/26144372793?pr=1852
- Artifact available on summary page:: https://github.com/NVIDIA/cccl/actions/runs/9487515807?pr=1852

Will need to rebase once #1846 is merged.

**Comment by alliepiper** [comment]
Waiting for #1844 to merge before rebasing.

File: .github/actions/docs-build/action.yml
Comment (wmaxey) [inline .github/actions/docs-build/action.yml:38]: Remove this part after the 404 page was integrated to Sphinx.

```suggestion
```

**Review by jrhemstad — APPROVED** [review]

File: .github/actions/docs-build/action.yml
Comment (alliepiper) [reply .github/actions/docs-build/action.yml:38]: Ha, I was just on my way to slack to ask you about this :)

**Review by wmaxey — APPROVED** [review]

**Comment by bdice** [comment]
Does this PR fix the failure shown here? https://github.com/NVIDIA/cccl/actions/runs/9488059535/job/26146095669#step:5:17

Something unexpected must have happened because I thought the 404 page should've worked from #1846... 

```
▶️ Run mkdir _site
  mkdir _site
  cp -rf ./docs/_build/docs/cccl/latest/* _site
  mkdir _site/cub
  cp -rf ./docs/_build/docs/cub/latest/* _site/cub
  mkdir _site/libcudacxx
  cp -rf ./docs/_build/docs/libcudacxx/latest/* _site/libcudacxx
  mkdir _site/thrust
  cp -rf ./docs/_build/docs/thrust/latest/* _site/thrust
  ./docs/scrape_docs.bash ./_site
  cp ./docs/404.html _site/404.html
  shell: /usr/bin/bash -e {0}
  env:
    GITHUB_PAGES: true
cp: cannot stat './docs/404.html': No such file or directory
Error: Process completed with exit code 1.
```

**Comment by wmaxey** [comment]
> Does this PR fix the failure shown here? https://github.com/NVIDIA/cccl/actions/runs/9488059535/job/26146095669#step:5:17
> 
> Something unexpected must have happened because I thought the 404 page should've worked from #1846...
> 
> ```
> ▶️ Run mkdir _site
>   mkdir _site
>   cp -rf ./docs/_build/docs/cccl/latest/* _site
>   mkdir _site/cub
>   cp -rf ./docs/_build/docs/cub/latest/* _site/cub
>   mkdir _site/libcudacxx
>   cp -rf ./docs/_build/docs/libcudacxx/latest/* _site/libcudacxx
>   mkdir _site/thrust
>   cp -rf ./docs/_build/docs/thrust/latest/* _site/thrust
>   ./docs/scrape_docs.bash ./_site
>   cp ./docs/404.html _site/404.html
>   shell: /usr/bin/bash -e {0}
>   env:
>     GITHUB_PAGES: true
> cp: cannot stat './docs/404.html': No such file or directory
> Error: Process completed with exit code 1.
> ```
> 
> edit: I verified the new docs artifacts from this PR and `404.html` definitely exists. I hope this fixes the problem.

Yeah, I addressed that.

**Pull request merged**
